src/utils/utils.cpp

```diff
@@ -111,6 +110,14 @@ std::string Escape(const std::string &src) {
   return ret;
 }
 
+template <typename T>
+inline void PrintIfNotZero(std::ostream &os, T value,
```

Comment:
Why is this `inline`? Also, maybe consider enabling this only if `T` is convertible to a bool?

Reply:
Will actually check if the type is arithmetic, thx for the suggestion.